tests/schema/test_schema.py

```diff
@@ -0,0 +1,251 @@
+    warnings = [part for part in warnings_raw.split("WARNING: ") if part]
+
+    expected_warnings = [
+        (
+            "'FEAt' does not match '^[A-Z0-9_]+$'",
+            "[sn_schema.local_fail]",
+        ),
+        (
+            "Unevaluated properties are not allowed ('asil', 'priority' were unexpected)",
+            "[sn_schema.local_fail]",
+        ),
+        (
+            "'approved' is a required property",
+            "[sn_schema.local_fail]",
+        ),
+        (
+            "'SPEC' does not match '^SPEC_[a-zA-Z0-9_-]*$'",
+            "[sn_schema.local_fail]",
+        ),
+        (
+            "Unevaluated properties are not allowed ('approved', 'asil', 'links', 'priority' were unexpected)",
+            "[sn_schema.local_fail]",
+        ),
+        (
+            "Too few valid links of type 'links' (0 < 1) / nok: FEAT",
+            "[sn_schema.network_contains_too_few]",
+        ),
+        (
+            "Unevaluated properties are not allowed ('approved', 'asil', 'links', 'priority' were unexpected)",
+            "[sn_schema.local_fail]",
+        ),
+        (
+            "Unevaluated properties are not allowed ('approved', 'asil', 'links', 'priority' were unexpected)",
+            "[sn_schema.local_fail]",
+        ),
+    ]
+    for expected in expected_warnings:
+        assert any(expected[0] in warning for warning in warnings), (
+            f"Expected warning not found: {expected[0]}"
+        )
+        if sphinx.version_info[0] >= 8:
+            assert any(expected[1] in warning for warning in warnings), (
+                f"Expected subtype not found: {expected[1]}"
+            )
+
+    assert len(warnings) == len(expected_warnings)
+    unexpected_warnings = [
+        '"approved" is a required property [sn_schema.validation_fail]',  # severity info too low
+    ]
+    for unexpected in unexpected_warnings:
+        assert all(unexpected not in warning for warning in warnings), (
+            f"Unexpected warning found: {unexpected}"
+        )
+
+    html = Path(test_app.outdir, "index.html").read_text()
+    assert html
+
+
+# split snapshot dir as warning (sub-)types are emitted only for Sphinx >= 8
+SNAPSHOT_DIR = (
+    "__snapshots__sphinx_lt_8"
+    if sphinx.version_info[0] < 8
+    else "__snapshots__sphinx_ge_8"
+)
+"""Snapshot directory name."""
+
+
+class DifferentDirectoryExtension(AmberSnapshotExtension):
+    """Overwrite the directory name for the snapshot files."""
+
+    @classmethod
+    def dirname(cls, *, test_location: "PyTestLocation") -> str:
+        return str(Path(test_location.filepath).parent.joinpath(SNAPSHOT_DIR))
+
+
+@pytest.fixture
+def snapshot(snapshot: syrupy.SnapshotAssertion):
+    return snapshot.use_extension(DifferentDirectoryExtension)
+
+
+@pytest.mark.parametrize(
+    "need_cnt",
+    [
+        10,
+        100,
+        1_000,
+        # 5_000,
+        # 10_000,
+    ],
+)
+@pytest.mark.benchmark
```

Comment:
there is a `tests/benchmark` folder, which this should go in

Reply:
Done